overlord/ifacestate/ifacestate.go

```diff
@@ -170,8 +170,8 @@ func Disconnect(st *state.State, plugSnap, plugName, slotSnap, slotName string)
 
 // CheckInterfaces checks whether plugs and slots of snap are allowed for installation.
 func CheckInterfaces(st *state.State, snapInfo *snap.Info) error {
-	// XXX: AddImplicitSlots is really a brittle interface
-	snap.AddImplicitSlots(snapInfo)
+	// XXX: addImplicitSlots is really a brittle interface
+	addImplicitSlots(snapInfo)
```

Comment:
can't we try to address the problem that is still brittle given that we are changing things ?